runtime/gc_modron_startup/mgcalloc.cpp

```diff
@@ -469,6 +469,11 @@ J9AllocateObject(J9VMThread *vmThread, J9Class *clazz, uintptr_t allocateFlags)
 	}	
 #endif /* J9VM_GC_THREAD_LOCAL_HEAP */	
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	if (J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassContainsUnflattenedFlattenables)) {
```

Comment:
need to make sure `objectPtr` is not null before doing this, also remove the ifdefs here